src/note.rs

```diff
@@ -209,18 +209,31 @@ impl Note {
         sponge_hash(&[sk_r, pos])
     }
 
-    /// Return a hash represented by `H(value_commitment, pos, H([R]),
-    /// H([pskr]))`
+    /// Return a hash represented by `H(note_type, value_commitment,
+    /// H(StealthAddress), pos, encrypted_data)
     pub fn hash(&self) -> BlsScalar {
         let value_commitment = self.value_commitment().to_hash_inputs();
         let pk_r = self.stealth_address().pk_r().to_hash_inputs();
+        let R = self.stealth_address().R().to_hash_inputs();
+        // The cipher is made being used differently to the
+        // other variables as we here enforce the PoseidonCipher
+        // wihtin each note to contain three scalars. This is a
+        // subsitution for an assert! method.
```

Comment:
I would remove the comment from here. Just add before the `sponge_hash` something like "// we assume cipher contains three scalars, this could change" or similar.